src/main/java/de/tum/in/www1/artemis/service/TextSubmissionService.java

```diff
@@ -235,10 +244,19 @@ public TextSubmission save(TextSubmission textSubmission) {
      *
      * @param exerciseId - the id of the exercise we are looking for
      * @param tutor - the tutor we are interested in
+     * @param examMode - flag should be set to ignore the test run submissions
      * @return a list of text Submissions
      */
-    public List<TextSubmission> getAllTextSubmissionsAssessedByTutorWithForExercise(Long exerciseId, User tutor) {
-        List<Submission> submissions = this.submissionRepository.findAllByParticipationExerciseIdAndResultAssessor(exerciseId, tutor);
+    public List<TextSubmission> getAllTextSubmissionsAssessedByTutorWithForExercise(Long exerciseId, User tutor, boolean examMode) {
+        List<Submission> submissions;
+        if (examMode) {
+            var participations = this.studentParticipationRepository.findAllByParticipationExerciseIdAndResultAssessorIgnoreTestRuns(exerciseId, tutor);
+            submissions = participations.stream().filter(studentParticipation -> studentParticipation.findLatestSubmission().isPresent())
+                    .map(StudentParticipation::findLatestSubmission).map(Optional::get).collect(toList());
```

Comment:
First map, then filter for present.
Also, check with similar comments above.

Reply:
See ad133719f0f0c316287bfa038a00989c9bd3b7fe and 781a856380be212ce234c33771cbcc231572e4c1